File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.py
from collections import Counter

def solution(data, col, row_begin, row_end):
    sum_xor = 0
    data.sort(key=lambda x:(x[col-1],-x[0]))

    mod_counter = Counter()
    for i in range(row_begin-1, row_end):
        row_mod = 0
        for c in range(len(data[0])):
            row_mod += data[i][c] % (i+1)
        mod_counter[row_mod] += 1

    for mod_val, count in mod_counter.items():
        if count % 2 != 0:
            sum_xor ^= mod_val
    
    return sum_xor

File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.js
function solution(data, col, row_begin, row_end) {
    const sortedData = sortData(data,col-1);
    const sliceData = sortedData.slice(row_begin-1,row_end);
    
    const result = sliceData.map((tp,i)=>
        tp.reduce((ac,cur)=> (cur%(row_begin+i)) + ac ,0)
    )
    
    return result.reduce((ac,cur)=> ac ^ cur , 0);
}

function sortData(data, targetCol){
    return data.sort((a,b)=> {
        if(a[targetCol] === b[targetCol]){
            return b[0]- a[0];
        }
        return a[targetCol] - b[targetCol];
    })
    
}